src/code/z_rcp.c

```diff
@@ -847,28 +847,50 @@ Gfx gEmptyDL[] = {
     gsSPEndDisplayList(),
 };
 
+/**
+ * Set fog color and range.
+ *
+ * At or prior to fog near, geometry is unaffected by fog. At or beyond fog far, geometry is fully fogged.
+ * Between near and far pixels will be linearly interpolated between the unfogged color and the supplied fog color.
+ *
+ * Fog far should be in the range 0 to 1000 and greater than or equal to fog near. If fog near is negative everything
+ * will be fully fogged. If fog near is 1000 or greater there is no fog.
+ */
 Gfx* Gfx_SetFog(Gfx* gfx, s32 r, s32 g, s32 b, s32 a, s32 near, s32 far) {
+    // Ensure fog far is greater than near
     if (far == near) {
         far++;
     }
-
     ASSERT(near != far, "n != f", "../z_rcp.c", 1155);
 
+    // Set the fog color, far away geometry will be rendered as this solid color.
     gDPSetFogColor(gfx++, r, g, b, a);
 
     if (near >= 1000) {
+        // Fog far is expected to be at most 1000 so near >= far. Set a constant shade alpha of 0 for no fog
```

Comment:
```suggestion
        // Set a constant shade alpha of 0 for no fog
```
I wouldn't write anything on why, this whole function is sloppy without good "why"s anyway